Tinyxml2Serializer.cpp

```diff
@@ -293,1133 +294,923 @@ CTinyxml2Deserializer::CTinyxml2Deserializer(const char* pData, std::uint32_t pM
     oXML.Parse(pData);
 }
 
-bool CTinyxml2Deserializer::DeserializeGeneralSettings(SSettingsGeneral& pGeneralSettings)
+namespace 
 {
-    //std::string             tStr;
 
-    //pGeneralSettings.vsCameras.clear();
+    bool ReadElementFloat(tinyxml2::XMLElement& element, const char* elementName, float& output)
+    {
+        if (auto childElem = element.FirstChildElement(elementName))
+        {
+            return childElem->QueryFloatText(&output) == tinyxml2::XML_SUCCESS;
+        }
 
-    //// ==================== General ====================
-    //if (!oXML.FindChildElem("General"))
-    //{
-    //    return false;
-    //}
-    //oXML.IntoElem();
+        return false;
+    }
 
-    //if (!oXML.FindChildElem("Frequency"))
-    //{
-    //    return false;
-    //}
-    //pGeneralSettings.nCaptureFrequency = atoi(oXML.GetChildData().c_str());
+    bool ReadElementUnsignedInt32(tinyxml2::XMLElement& element, const char* elementName, std::uint32_t& output)
+    {
+        if (auto childElem = element.FirstChildElement(elementName))
+        {
+            return childElem->QueryUnsignedText(&output) == tinyxml2::XML_SUCCESS;
+        }
 
-    //if (!oXML.FindChildElem("Capture_Time"))
-    //{
-    //    return false;
-    //}
-    //pGeneralSettings.fCaptureTime = (float)atof(oXML.GetChildData().c_str());
+        return false;
+    }
 
-    //// Refactored variant of all this copy/paste code. TODO: Refactor everything else.
-    //if (!ReadXmlBool(&oXML, "Start_On_External_Trigger", pGeneralSettings.bStartOnExternalTrigger))
-    //{
-    //    return false;
-    //}
-    //if (mnMajorVersion > 1 || mnMinorVersion > 14)
-    //{
-    //    if (!ReadXmlBool(&oXML, "Start_On_Trigger_NO", pGeneralSettings.bStartOnTrigNO))
-    //    {
-    //        return false;
-    //    }
-    //    if (!ReadXmlBool(&oXML, "Start_On_Trigger_NC", pGeneralSettings.bStartOnTrigNC))
-    //    {
-    //        return false;
-    //    }
-    //    if (!ReadXmlBool(&oXML, "Start_On_Trigger_Software", pGeneralSettings.bStartOnTrigSoftware))
-    //    {
-    //        return false;
-    //    }
-    //}
+    bool ReadElementStringAllowEmpty(tinyxml2::XMLElement& element, const char *elementName,std::string& output )
```

Comment:
It's a minor thing, but the asterisk is on the wrong side: char *element --> **char* element**